### PR TITLE
#174 Add / Delete Trips and Fixes

### PR DESCRIPTION
**Pull request by morgankbender**
Morgan and Graeme adding and fixing things

### CODE REVIEW

File: src/components/auth/signup/SignUp.js [hunk 1]
```diff
@@ -25,17 +25,14 @@ export default class SignUp extends Component {
   /**************************** Helper functions ****************************/
 
   stateIsComplete() {
-    if (
-      (this.state.f_name,
+    return (
```
Comment (evscott) [inline src/components/auth/signup/SignUp.js]: Nice.

File: src/components/dashboard/DashboardNav.js [hunk 2]
```diff
@@ -11,15 +12,41 @@ export default class DashboardNav extends Component {
...
+  quitCreateEvent() {
+    this.setState({ showNewTrip: false });
+  }
+
+  /**
+   * This opens the create trip modal.
+   * @param e click event
+   */
+  startCreateEvent(e) {
+    e.preventDefault();
```
Comment (evscott) [inline src/components/dashboard/DashboardNav.js]: Out of curiosity, is there a reason we use preventDefault in some places but don't in others?

File: src/components/dashboard/NewTripForm.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import { Button, Form } from 'react-bootstrap';
+import { FormInput } from '../global/FormInput';
+
+export default class NewTripForm extends Component {
```
Comment (evscott) [inline src/components/dashboard/NewTripForm.js]: Very clean!

File: src/components/dashboard/NewTripFormContainer.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { connect } from 'react-redux';
+import NewTripForm from './NewTripForm';
+import { addTripToDb } from '../../redux/actions/tripActions';
+
+const mapStateToProps = (state, ownProps) => {
+  return {
+    close: ownProps.close
```
Comment (evscott) [inline src/components/dashboard/NewTripFormContainer.js]: I haven't seen this `ownProps` before. This is a method, though? My understanding is that it closes the modal which a component is contained in?

File: src/components/dashboard/PreviousTrips.js [hunk 5]
```diff
@@ -23,6 +33,13 @@ export default class PreviousTrips extends Component {
           <p>{trip.name}</p>
         </Link>
         <p>{trip.dscript}</p>
+        <Button
+          variant={'danger'}
+          className={'float-right'}
+          onClick={() => this.handleDelete(trip)}
+        >
+          <FontAwesomeIcon size={'sm'} icon={['fas', 'bomb']} />
```
Comment (evscott) [inline src/components/dashboard/PreviousTrips.js]: 💣💣💣

File: src/components/dashboard/UpcomingTrips.js [hunk 6]
```diff
@@ -23,6 +33,13 @@ export default class UpcomingTrips extends Component {
           <p>{trip.name}</p>
         </Link>
         <p>{trip.dscript}</p>
+        <Button
+          variant={'danger'}
+          className={'float-right'}
+          onClick={() => this.handleDelete(trip)}
```
Comment (evscott) [inline src/components/dashboard/UpcomingTrips.js]: 💣💣💣

Side note - thoughts on abstracting this delete button into its own class? We could then reuse it for all of our delete functionality. My thought is that we could place this in our `global` folder just like the return, submit, and edit button. (Doesn't need to be done in this pull request, of course).

File: src/redux/actions/tripActions.js [hunk 7]
```diff
@@ -313,6 +314,7 @@ export function updateTripInDb(trip) {
  */
 export function deleteTripInDb(trip) {
   return (dispatch) => {
+    console.log(trip);
```
Comment (evscott) [inline src/redux/actions/tripActions.js]: I assume this shouldn't be here :).

File: src/redux/actions/tripActions.js [hunk 8]
```diff
@@ -262,6 +262,7 @@ export function addTripToDb(trip) {
       .then((json) => {
         if (json.success === false) dispatch(addTripToDbFailure());
         else {
+          console.log(json.result);
```
Comment (evscott) [inline src/redux/actions/tripActions.js]: I assume this shouldn't be here :).

File: src/redux/reducers/tripReducer.js [hunk 9]
```diff
@@ -123,11 +117,9 @@ export function tripReducer(state = initialState, action) {
         lastUpdated: action.lastUpdated,
         isFetching: action.isFetching,
         isSynced: action.isSynced,
-        trips: filterSingleDimensionalArray(
-          state.trips,
-          action.tripToDelete.trip_id,
-          action.tripToDelete.e_id
-        )
+        trips: state.trips.filter((trip) => {
+          return trip.trip_id !== action.tripToDelete.trip_id;
```
Comment (evscott) [inline src/redux/reducers/tripReducer.js]: Nice 👍.

**Review by evscott — DISMISSED** [review]
Good stuff! Just need to remove two logs.

File: src/components/dashboard/NewTripFormContainer.js (hunk 4 above)
Comment (morgankbender) [reply src/components/dashboard/NewTripFormContainer.js]: ownProps is just the typical style referring to props passed to a container (other than the state). In this case yupp, the method to close the modal

File: src/components/dashboard/DashboardNav.js (hunk 2 above)
Comment (morgankbender) [reply src/components/dashboard/DashboardNav.js]: Tbh I might over use it, but I tend to call it whenever I'm creating a click action. It just keeps the browser from doing anything you haven't directly specified is my understanding

File: src/components/dashboard/UpcomingTrips.js (hunk 6 above)
Comment (morgankbender) [reply src/components/dashboard/UpcomingTrips.js]: Hmm, this might be a good idea if we get a chance

**Review by evscott — APPROVED** [review]

**Pull request merged**